### PR TITLE
Replace phantom.js with chrome headless (puppeteer)

### LINKED ISSUE #473: Replace PhantomJS with headless Chrome
We have a [script](https://github.com/styleguidist/react-styleguidist/blob/master/test/phantom.js) that loads a production build of a style guide and checks that there’s no JavaScript errors. We run in [on Travis CI](https://github.com/styleguidist/react-styleguidist/blob/master/.travis.yml#L46-L53) to check that production bundles actually work.

Now the script uses PhantomJS which has some issues like always out of date WebKit version and most likely it will [not be supported](https://groups.google.com/forum/#!topic/phantomjs/9aI5d-LDuNE) anymore. 

The new [headless Chrome](https://developers.google.com/web/updates/2017/04/headless-chrome) looks like a better and future friendly alternative.

There are wrappers like [](https://github.com/OnetapInc/chromy) that should make interaction with Chrome easier.

### PR DESCRIPTION
**Pull request by devsnek**
moves `phantom.js` to `browser.js` with the same functionality

closes #473

### CODE REVIEW

**Comment by sapegin** [comment]
Thanks, that‘s awesome!

I think we should run it only on Node 8:

https://github.com/styleguidist/react-styleguidist/blob/5ad09e4faf7262a97fc6be13c4718551e8c08f7d/.travis.yml#L32

That way we could continue using async/await and will make CI faster.

**Comment by devsnek** [comment]
I'll throw that in when I get home

**Comment by sapegin** [comment]
Almost ;-| Puppeteer doesn’t support Node 4, so we’ll have to remove it from devDependencies and install on CI, like this:

```
npm install --no-save puppeteer
```

before running it.

File: test/browser.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	page.on('console', (...args) => console.log('PAGE LOG:', ...args));
+
+	const url = /https?/.test(args[0]) ? args[0] : `file://${path.resolve(args[0])}`;
+	await page.goto(url);
+
+	if (args[1]) {
+		await page.screenshot({ path: args[1] });
+	}
+
+	await browser.close();
```
Comment (vlad-zhukov) [inline test/browser.js]: If it throws above the browser will not be properly closed.

**Comment by sapegin** [comment]
There’s one more lint issue ;-|

And we’ll need to add `|| exit 1` after each command to short circuit them. (May not be part of this PR.)

**Comment by sapegin** [comment]
Could you also remove `travis_phantomjs` from cache folders in `.travis.yml`?

**Comment by sapegin** [comment]
Thanks, merging! 🍕

**Pull request merged**